### PR TITLE
migrate to faraday

### PR DESCRIPTION
**Pull request by eserra**

### CODE REVIEW

File: frontapp.gemspec [hunk 1]
```diff
@@ -15,7 +15,7 @@ Gem::Specification.new do |s|
   s.files       += Dir.glob("lib/frontapp/utils/*.rb")
   s.homepage    = 'https://github.com/phusion/frontapp'
   s.license     = 'MIT'
-  s.add_dependency 'http', '>= 5'
+  s.add_dependency 'faraday'
```
Comment (wyattades) [inline frontapp.gemspec]: let's lock the faraday major version

File: lib/frontapp/client.rb [hunk 2]
```diff
@@ -43,7 +43,7 @@ class Client
     def initialize(options={})
       auth_token = options[:auth_token]
       user_agent = options[:user_agent] || "Frontapp Ruby Gem #{VERSION}"
-      @headers = HTTP.headers({
+      @connection = Faraday.new(headers: {
```
Comment (wyattades) [inline lib/frontapp/client.rb]: can also add base_url here

File: lib/frontapp/client.rb [hunk 3]
```diff
@@ -56,11 +56,9 @@ def list(path, params = {})
       query = format_query(params)
       url = "#{base_url}#{path}?#{query}"
       until last_page
-        res = @headers.get(url)
-        if !res.status.success?
-          raise Error.from_response(res)
-        end
-        response = JSON.parse(res.to_s)
+        res = @connection.get(url)
+        raise Error.from_response(res) unless res.status == 200
```
Comment (wyattades) [inline lib/frontapp/client.rb]: (res.status / 100.0).floor == 2

**Review by wyattades — CHANGES_REQUESTED** [review]

File: lib/frontapp/client.rb (hunk 3 above)
Comment (wyattades) [reply lib/frontapp/client.rb]: actually, there appears to be a `success?` helper https://lostisland.github.io/faraday/#/getting-started/env-object?id=helpers

Comment (eserra) [reply lib/frontapp/client.rb]: I tried locally, but it did not work, let me have a look

Comment (eserra) [reply lib/frontapp/client.rb]: `res.status.to_s.starts_with?("2")` :)

Comment (eserra) [reply lib/frontapp/client.rb]: ok... found the helper

File: lib/frontapp/client.rb (hunk 2 above)
Comment (eserra) [reply lib/frontapp/client.rb]: done

File: lib/frontapp/client.rb [hunk 4]
```diff
@@ -73,59 +73,43 @@ def list(path, params = {})
     end
 
     def get(path)
-      res = @headers.get("#{base_url}#{path}")
-      if !res.status.success?
-        raise Error.from_response(res)
-      end
-      JSON.parse(res.to_s)
+      res = @connection.get("#{path}")
```
Comment (wyattades) [inline lib/frontapp/client.rb]:
```suggestion
      res = @connection.get(path)
```

File: lib/frontapp/client.rb [hunk 5]
```diff
@@ -43,24 +43,24 @@ class Client
...
+          "User-Agent": user_agent
+        })
     end
 
     def list(path, params = {})
       items = []
       last_page = false
       query = format_query(params)
-      url = "#{base_url}#{path}?#{query}"
+      url = "#{path}?#{query}"
```
Comment (wyattades) [inline lib/frontapp/client.rb]:
```suggestion
      url = query.empty? ? path : "#{path}?#{query}"
```

**Review by wyattades — APPROVED** [review]
couple nitpicks

File: lib/frontapp/client.rb (hunk 4 above)
Comment (eserra) [reply lib/frontapp/client.rb]: ahah, right :)

File: lib/frontapp/client.rb [hunk 6]
```diff
@@ -73,59 +73,43 @@ def list(path, params = {})
...
     end
 
     def get_plain(path)
-      headers_copy = @headers.dup
-      res = @headers.accept("text/plain").get("#{base_url}#{path}")
-      if !res.status.success?
-        raise Error.from_response(res)
-      end
-      res.to_s
+      res = @connection.get("#{path}", headers: {Accept: "text/plain"})
```
Comment (wyattades) [inline lib/frontapp/client.rb:82]: there's a few other examples of this

**Pull request merged**